### PR TITLE
Sid/like dislike functionality main

### PR DESCRIPTION
**Pull request by sid-s1**
Hey guys,

This PR includes like-dislike functionality; inclusions are -
* Controller routes and Model DB calls for checking current like/dislike state, creating like/dislike based on which button user clicks, editing like/dislike based on what they click, un-liking/un-disliking functionality, remembering the responses
* Ensuring single-presses work and everytime one of those buttons is clicked, the button fills up to show if the post is liked or disliked and at the same time the info is stored/updated in the database
* Everything working well in conjunction - reading, creating, updating, deleting - without needing to reload the whole page; only reloading the containers for the likes and dislikes based on button presses and API calls

Started work on counting votes so that likes and dislikes can be displayed below the respective buttons; there are some issues here that will need to be addressed - I will work on them once we have finished deployment today.

Thanks!

### CODE REVIEW

File: models/like-dislike.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const db = require('../database/db');
+
+const likeDislike = {
```
Comment (Chrispy1987) [inline models/like-dislike.js:3]: Hi Sid, could you please consider changing this object/class name to 'Votes' (or another single name that helps describe what this object/class does)  as `likeDislike.likeDislike` doesn't read very well

**Review by Chrispy1987 — CHANGES_REQUESTED** [review]
Change requested - Please refer to comment on the models/like-dislike.js file

Thanks Sid!

File: models/like-dislike.js (hunk 1 above)
Comment (sid-s1) [reply models/like-dislike.js]: Done! Thanks Chris!

**Review by Chrispy1987 — APPROVED** [review]
Excellent work Sid! Love that functionality and code looks SOLID :)

**Pull request merged**